src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/NotifyService.kt

```diff
@@ -8,32 +8,63 @@ import org.springframework.web.util.UriComponentsBuilder
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEvent
 import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.events.ReferralEventType
 import uk.gov.service.notify.NotificationClient
+import java.net.URI
+import java.util.UUID
 
 @Service
 class NotifyService(
   @Value("\${notify.enabled}") private val enabled: Boolean,
   @Value("\${notify.templates.referral-sent}") private val referralSentTemplateID: String,
+  @Value("\${notify.templates.referral-assigned}") private val referralAssignedTemplateID: String,
   @Value("\${interventions-ui.baseurl}") private val interventionsUIBaseURL: String,
   @Value("\${interventions-ui.locations.sent-referral}") private val interventionsUISentReferralLocation: String,
   private val client: NotificationClient,
+  private val hmppsAuthService: HMPPSAuthService,
 ) : ApplicationListener<ReferralEvent> {
   override fun onApplicationEvent(event: ReferralEvent) {
```

Comment:
_observation_
🤔 this method would now probably trip a cyclomatic complexity or function lines-of-code static check

do we want those to run on the code? (I'm not suggesting to rewrite, just interested in if we want to setup something like that)

Reply:
good point. yes i'd be up for running something like this. rajiv is adding to this class as we speak so he could include some refactoring there? @mahalrmoj